### DISCUSSION TITLE
Add a custom language to spacy

### DISCUSSION THREAD

**Discussion by Mujerry** [explosion/spaCy#10827] [Language Support]
I am trying to add a custom langauge(Luganda from Uganda) to spacy and am facing difficulties to use the provided sources. so far i have the __init__.py file ,stopwords and lexical attributes file ready but i cannot figure out how to add it to spacy. Below is the __inti__.py file 
```
import spacy
from spacy.language import Language
from stop_words import STOP_WORDS

class LugandaDefaults(Language.Defaults):
    stop_words = STOP_WORDS

@spacy.registry.languages("lg")
class Luganda(Language):
    """docstring for ."""
    lang="lg"
    Defaults=LugandaDefaults
__all__=["Luganda"]
```

**Comment by polm** [comment]
Sorry you're having trouble with this. 

To be clear, do you want to create this language for your own use, or add it to the main spaCy repo? What were you specifically unable to do, and did you get an error?

If you just want to use the language in your program, your code should be mostly fine, you just don't need the `__all__` line - that's just dealing with module organization. I don't think it's hurting anything though. 

If you want to add it to spaCy directly, in general the right way to add a language is to copy an existing one that's similar and modify the files as required. In this case you could put your source files in a directory called `lg` under `spacy/lang`, along with the existing languages. You'll also need to modify your imports a bit - if you open a PR we could help you with that.

**Reply by tobiusaolo** [reply]
Thank you a lot , I copied Yoruba language and after that, I cloned the spacy repo , and added my lg to spacy/lang but they prompt permission denied when I try to push to the repo and when I use the GitHub UI it shows uploads require push access to the repository

**Reply by polm** [reply]
Are you the same user?

If you want to add to the spaCy repo you'll need to submit a "pull request" so we can check your code and say it's OK. That's so not just anybody can modify the code. 

Github has a guide to pull requests [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request). If you push to your own fork of the repo a button should show up in the Github UI.

**Reply by tobiusaolo** [reply]
Alright then, let me work on that and I get back to you. Once again thank you for the quick responses
